app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/HotspotService.java

```diff
@@ -0,0 +1,151 @@
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.main.MainActivity;
+import org.kiwix.kiwixmobile.utils.Constants;
+import org.kiwix.kiwixmobile.webserver.ServerStateListener;
+
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_TURN_OFF_AFTER_O;
+import static org.kiwix.kiwixmobile.webserver.ZimHostActivity.ACTION_TURN_ON_AFTER_O;
+import static org.kiwix.kiwixmobile.webserver.WebServerHelper.stopAndroidWebServer;
+
+/**
+ * HotspotService is used to add a foreground service for the wifi hotspot.
+ * Created by Adeel Zafar on 07/01/2019.
+ */
+
+public class HotspotService extends Service {
+  private static final int HOTSPOT_NOTIFICATION_ID = 666;
+  public static final String ACTION_STOP = "hotspot_stop";
+  public static WifiHotspotManager hotspotManager;
+  private BroadcastReceiver stopReceiver;
+  private NotificationManager notificationManager;
+  private NotificationCompat.Builder builder;
+  ServerStateListener serverStateListener;
+  String TAG = HotspotService.this.getClass().getSimpleName();
+
+  @Override public void onCreate() {
+    super.onCreate();
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      stopReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+          if (intent != null && intent.getAction().equals(ACTION_STOP)) {
+            stopHotspot();
+          }
+        }
+      };
+    }
+    registerReceiver(stopReceiver, new IntentFilter(ACTION_STOP));
+    notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+    startForeground(HOTSPOT_NOTIFICATION_ID,
+        buildForegroundNotification(getString(R.string.hotspot_start), false));
+  }
+
+  @Override public int onStartCommand(Intent intent, int flags, int startId) {
+    switch (intent.getAction()) {
+
+      case ACTION_TURN_ON_AFTER_O:
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          hotspotManager.turnOnHotspot();
+          updateNotification(getString(R.string.hotspot_running), true);
+        }
+        break;
+
+      case ACTION_TURN_OFF_AFTER_O:
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          stopHotspot();
+        }
+        break;
+      default:
+        break;
+    }
+    return START_NOT_STICKY;
+  }
+
+  @Nullable @Override public IBinder onBind(Intent intent) {
+    return null;
+  }
+
+  private Notification buildForegroundNotification(String status, boolean showStopButton) {
+    Log.v(TAG, "Building notification " + status);
+    builder = new NotificationCompat.Builder(this);
+    builder.setContentTitle("Kiwix Hotspot").setContentText(status);
+    Intent targetIntent = new Intent(this, MainActivity.class);
+    targetIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
```

Comment:
Does this work? Usually when starting an activity using a Service context you need the `NEW_TASK` flag

Reply:
Fixed. 05d7546c9c6582e6d8bae42969fb9699aad10a17